### PR TITLE
Rename sendPayload to sendMessage

### PR DESCRIPTION
**Pull request by carl-mastrangelo**
Addresses issue #637

### CODE REVIEW

File: interop-testing/src/main/java/io/grpc/testing/integration/AbstractTransportTest.java [hunk 1]
```diff
@@ -711,7 +711,7 @@ public void serviceAccountCreds(String jsonKey, InputStream credentialsStream, S
       throws Exception {
     // cast to ServiceAccountCredentials to double-check the right type of object was created.
     GoogleCredentials credentials =
-        (ServiceAccountCredentials) GoogleCredentials.fromStream(credentialsStream);
+        GoogleCredentials.fromStream(credentialsStream);
```
Comment (carl-mastrangelo) [inline interop-testing/src/main/java/io/grpc/testing/integration/AbstractTransportTest.java]: Whoops again, reverted.

**Comment by madongfly** [comment]
@carl-mastrangelo  LGTM

**Comment by ejona86** [comment]
Doing the `UnsupportedOperationException` effectively kills any API compatibility that was done for the PR. It would have been better to remove `sendPayload` outright than start throwing an `UnsupportedOperationException`.

**Comment by carl-mastrangelo** [comment]
Are you sure?  All existing users override it.  All new users will have to override sendMessage

**Comment by ejona86** [comment]
"All existing users override it"? What do you mean? It doesn't appear that any code overrides it. `ClientCallImpl` at the very least does not override `sendPayload()`. That means that any existing users of `sendPayload()` will compile, but not work. It seems it would have been better to either fail the compile (by removing `sendPayload()`) or to make them work (by having `sendPayload()` call `sendMessage()`).

**Comment by carl-mastrangelo** [comment]
```
class OldWay extends ClientCall {
  @Override
  public onPayload(T payload) {
    ....
  }
}

class NewWay extends ClientCall {
  @Override
  public onMessage(T message) {
    ....
  }
}

class NeverSupportedWay extends ClientCall {
  
}
```


Old way works because children override onPayload, which was previously abstract.  ClientCall never throws.  In New way, children override onMessage, which results in ClientCall.onMessage never calling onPayload, so nothing throws.  If neither method is overridden, the calls to onMessage result in UOEs.  (previously a compiler error, now a runtime error, a small price for breaking api changes IMO).  Individual library users can switch from the old override to the new override safely and independently of each other.  Once all implementations of ClientCall are updated, onPayload can be safely removed.

**Comment by ejona86** [comment]
@carl-mastrangelo, you didn't change `onPayload`, you changed `sendPayload`.

**Comment by carl-mastrangelo** [comment]
See PR #684

**Pull request merged**